src/aggregate.d

```diff
@@ -270,13 +276,15 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
Not sure if it is the same ICE in GDC, but @kinke fixed it by not adding the same var declaration to the fields twice: https://github.com/ldc-developers/ldc/pull/1987/commits/1094634364e2977d53f9655c9b07627eb3e494ab